determined-ai/determined
# feat: partial checkpoint delete [DET-9491]

## Description

Allows partial checkpoint deletion through sdk or added cli command.

## Test Plan

unit + integration + e2e tests

manual testing

Create experiment
```
det e create examples/tutorials/mnist_pytorch/const.yaml examples/tutorials/mnist_pytorch/ -f
```

Get checkpoint id
```
det e lc $exp_id --csv
```

Partial delete no op
```
det checkpoint rm 663013e4-1234-476b-a151-6b7aa77ece87
```

Should see (wait for the task to finish by it falling off ```det task``` and use ```det e lc $exp_id --csv```)
```Trial ID,# of Batches,State,Validation Metric,UUID,Resources,Size
26068,937,COMPLETED,0.0497250406331161,663013e4-1234-476b-a151-6b7aa77ece87,"code/
code/README.md
code/adaptive.yaml
code/const.yaml
code/data.py
code/dist_random.yaml
code/distributed.yaml
code/layers.py
code/metrics_spam.py~
code/model_def.py
code/simple_script.py~
load_data.json
metadata.json
state_dict.pth
trial_state.pkl",13.8MB
```

Remove `load_data.json` manually from file system and start partial delete no op
```
rm /tmp/determined-cp/663013e4-1234-476b-a151-6b7aa77ece87/load_data.json
det checkpoint rm 663013e4-1234-476b-a151-6b7aa77ece87
```

Should see
```
Trial ID,# of Batches,State,Validation Metric,UUID,Resources,Size
26068,937,PARTIALLY_DELETED,0.0497250406331161,663013e4-1234-476b-a151-6b7aa77ece87,"code/
code/README.md
code/adaptive.yaml
code/const.yaml
code/data.py
code/dist_random.yaml
code/distributed.yaml
code/layers.py
code/metrics_spam.py~
code/model_def.py
code/simple_script.py~
metadata.json
state_dict.pth
trial_state.pkl",13.8MB
```

Remove code directory
```
det checkpoint rm 663013e4-1234-476b-a151-6b7aa77ece87 --glob "code/*"
```

Should see
```
Trial ID,# of Batches,State,Validation Metric,UUID,Resources,Size
26068,937,PARTIALLY_DELETED,0.0497250406331161,663013e4-1234-476b-a151-6b7aa77ece87,"code/
metadata.json
state_dict.pth
trial_state.pkl",13.8MB

```

Fully delete checkpoint
```
det checkpoint rm 663013e4-1234-476b-a151-6b7aa77ece87 --glob "trial_state.pkl" --glob "state_dict.pth" --glob "code" --glob "metadata.json"
```

Should see
```
Trial ID,# of Batches,State,Validation Metric,UUID,Resources,Size
26068,937,DELETED,0.0497250406331161,663013e4-1234-476b-a151-6b7aa77ece87,"code/
metadata.json
state_dict.pth
trial_state.pkl",13.8MB

```
## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
